WordPress/WordPressApi/WordPressOrgXMLRPCApi.swift

```diff
@@ -0,0 +1,163 @@
+import Foundation
+import wpxmlrpc
+
+public class WordPressOrgXMLRPCApi: NSObject
+{
+    public typealias SuccessResponseBlock = (responseObject: AnyObject, httpResponse: NSHTTPURLResponse?) -> ()
+    public typealias FailureReponseBlock = (error: NSError, httpResponse: NSHTTPURLResponse?) -> ()
+
+    private let endpoint: NSURL
+    private let userAgent: String?
+
+    private lazy var session: NSURLSession = {
+        let sessionConfiguration = NSURLSessionConfiguration.defaultSessionConfiguration()
+        var additionalHeaders: [String : AnyObject] = ["Accept-Encoding":"gzip, deflate"]
+        if let userAgent = self.userAgent {
+            additionalHeaders["User-Agent"] = userAgent
+        }
+        sessionConfiguration.HTTPAdditionalHeaders = additionalHeaders
+        let session = NSURLSession(configuration: sessionConfiguration, delegate: self, delegateQueue: nil)
+        return session
+    }()
+
+    public init(endpoint: NSURL, userAgent: String? = nil) {
+        self.endpoint = endpoint
+        self.userAgent = userAgent
+        super.init()
+    }
+
+    deinit {
+        session.finishTasksAndInvalidate()
+    }
+
+    /**
+     Cancels all ongoing and makes the session so the object will not fullfil any more request
+     */
+    public func invalidateAndCancelTasks() {
+        session.invalidateAndCancel()
+    }
+
+    //MARK: - Network requests
+
+    /**
+     Executes a XMLRPC call for the method specificied with the arguments provided.
+
+     - parameter method:  the xmlrpc method to be invoked
+     - parameter parameters: the parameters to be encoded on the request
+     - parameter success:    callback to be called on successful request
+     - parameter failure:    callback to be called on failed request
+
+     - returns:  a NSProgress object that can be used to track the progress of the request and to cancel the request. If the method
+     returns nil it's because something happened on the request serialization and the network request was not started, but the failure callback
+     will be invoked with the error specificing the serialization issues.
+     */
+    public func callMethod(method: String,
+                           parameters: [AnyObject]?,
+                           success: SuccessResponseBlock,
+                           failure: FailureReponseBlock) -> NSProgress?
+    {
+        //Encode request
+        var request: NSURLRequest? = nil
```

Comment:
You don't need to make this optional, you can have `let request: NSURLRequest` and as long as you set a value before using it the compiler will be happy.